DxDispatch/src/dxdispatch/OnnxDispatchable.cpp

```diff
@@ -522,4 +523,97 @@ void OnnxDispatchable::Dispatch(const Model::DispatchCommand& args, uint32_t ite
+
+                    Model::BufferDesc desc;
+                    desc.sizeInBytes = (bufferDesc.elementSizeInBytes * bufferDesc.elementCount);
+                    desc.initialValues = std::vector<std::byte>(
+                        tensorData,
+                        tensorData + desc.sizeInBytes);
+                    desc.initialValuesOffsetInBytes = 0;
+
+                    auto elementType = shapeInfo.GetElementType();
+                    if (ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_FLOAT16;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_FLOAT32;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_FLOAT64;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_INT8 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_INT8;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_INT16 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_INT16;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_INT32 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_INT32;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_INT64;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT8 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_UINT8;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT16 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_UINT16;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT32 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_UINT32;
+                    }
+                    else if (ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64 == elementType)
+                    {
+                        desc.initialValuesDataType = DML_TENSOR_DATA_TYPE_UINT64;
+                    }
+                    else
+                    {
+                        DebugBreak();
+                    }
+                    bufferDesc.resourceDesc->value = std::move(desc);
```

Comment:
It looks like `bufferDesc.resource` is still null, so the user can't print the contents of deferred resources. Isn't this the purpose of exposing deferred resources through the JSON model?

Reply:
I only saw values[i].GetTensorMutableRawData() this value as a tensor this appears to have the updated tensor data